pandas/tests/indexes/datetimes/test_datetime.py

```diff
@@ -153,6 +153,17 @@ def test_iteration_preserves_tz(self):
             assert result._repr_base == expected._repr_base
             assert result == expected
 
+    @pytest.mark.parametrize('periods', [0, 9999, 10000, 10001])
+    def test_iteration_over_chunksize(self, periods):
+        # GH21012
+
+        index = date_range('2000-01-01 00:00:00', periods=periods, freq='min')
+        num = 0
+        for i, d in enumerate(index):
+            assert index[num] == d
```

Comment:
Use `i` instead of `num`?

Reply:
I wanted to check boundary values including `periods == 0`.
Without `num`, it would be a bit complicated, isn't it ?

Reply:
That's fine then, though you can remove the `enumerate` since `i` isn't used anywhere I think.